### PR TITLE
set natgateway ip name in defaults

### PR DESCRIPTION
**Pull request by shysank**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note

```

### CODE REVIEW

File: azure/scope/cluster.go [hunk 1]
```diff
@@ -241,9 +230,10 @@ func (s *ClusterScope) NatGatewaySpecs() []azure.NatGatewaySpec {
 		natGateways = append(natGateways, azure.NatGatewaySpec{
 			Name: natGateway.Name,
 			NatGatewayIP: infrav1.PublicIPSpec{
-				Name: s.GetNodeNatGatewayIPName(),
+				Name: natGateway.NatGatewayIP.Name,
 			},
 			Subnet: s.NodeSubnet(),
+			// hello
```
Comment (CecileRobertMichon) [inline azure/scope/cluster.go:236]: 👋

Comment (shysank) [reply azure/scope/cluster.go:236]: oops sorry 😄

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve
/release-note-none

**Pull request merged**